### ISSUE TITLE
TypeError: ContextualIdentityService.getIdentityFromId(...) is undefined cookies.js:88:15

### ISSUE THREAD

**Issue by kjozwiak** [mozilla/testpilot-containers#172]
We're breaking the cookie manager when removing containers with associated tabs. I can reproduce this pretty reliably with custom containers. Once this happens, the user won't be able to view/manage their cookies via about:preferences#privacy..

See also [bug#1319929](https://bugzilla.mozilla.org/show_bug.cgi?id=1319929)

```
TypeError: ContextualIdentityService.getIdentityFromId(...) is undefined cookies.js:88:15
	gCookiesWindow._isPrivateCookie chrome://browser/content/preferences/cookies.js:88:15
	gCookiesWindow._loadCookies chrome://browser/content/preferences/cookies.js:503:13
	gCookiesWindow._populateList chrome://browser/content/preferences/cookies.js:52:5
	gCookiesWindow.init chrome://browser/content/preferences/cookies.js:35:5
	onload chrome://browser/content/preferences/cookies.xul:1:1
```

<img width="572" alt="screen shot 2017-02-10 at 4 19 05 pm" src="https://cloud.githubusercontent.com/assets/2602313/22844516/aa98ef2a-efac-11e6-97af-c1866bff16bd.png">

STR:

* install fx51.0.1, buildId: 20170125094131, changeset: 327e081221b0
* install the latest XPI (0.0.2) in this instance
* create a new customer container and open reddit.com
* open several of the top links in the new custom container via the right click context menu
* hide all the tabs that are opened in the custom container via "Hide these container tabs"
* once all the tabs are hidden, delete the entire container
* access the cookie manager via about:preferences#privacy

**Comment by TanviHacks** [comment]
Marking this as MVP.  Let me know if you disagree.

**Comment by groovecoder** [comment]
I was able to reproduce this in fx 51.0b14 on a blank profile without installing any add-ons at all. Is this a lower-level bug? Not related to this TP add-on/experiment?

**Comment by kjozwiak** [comment]
> I was able to reproduce this in fx 51.0b14 on a blank profile without installing any add-ons at all. Is this a lower-level bug? Not related to this TP add-on/experiment?

Correct! Unfortunately the bug is in the platform on all the current channels. I created [bug#1319929](https://bugzilla.mozilla.org/show_bug.cgi?id=1319929) which addresses this bug a few months ago but hasn't beed fixed yet. Baku has been working on a patch in [bug#1337964](https://bugzilla.mozilla.org/show_bug.cgi?id=1337964) that should address this issue.

With the current container add-on, it seems like the issue is a lot easier to reproduce due to ability to easily create/remove/change containers on the fly. The issue is that the fix in [bug#1337964](https://bugzilla.mozilla.org/show_bug.cgi?id=1337964) would have to be uplifted all the way into the current release build, which I don't think will happen.

We need to find a fix that we can apply to the add-on directly rather than waiting for [bug#1337964](https://bugzilla.mozilla.org/show_bug.cgi?id=1337964) to get fixed and make it's way through the channels.

**Comment by groovecoder** [comment]
I was able to reproduce this without creating, removing, or changing any containers at all.

**Comment by kjozwiak** [comment]
There's definitely several other ways of reproducing the issue.. I've managed to reproduce it via restoring sessions. Sometimes a restart will fix the issue, sometimes the cookie manager stays broken/unusable until you re-install start a new profile.

**Comment by groovecoder** [comment]
@bakulf - would be be able to uplift whatever patch you make for bug#1337964 into our ContextualIdentityService usage?

**Comment by bakulf** [comment]
This would not help: the addon will be released next week. We need to find a fix by our own :)

**Comment by groovecoder** [comment]
This seems like the last major issue blocking release. @bakulf - it's assigned to you? Any chance to work on it?

**Comment by bakulf** [comment]
Yes, I'm working on that. I'm trying hard to avoid to implement an horrible hack...
